src/coreclr/src/jit/importer.cpp

```diff
@@ -18715,8 +18685,7 @@ void Compiler::impCheckCanInline(GenTreeCall*           call,
             // trying to inline this method.
             initClassResult =
                 pParam->pThis->info.compCompHnd->initClass(nullptr /* field */, pParam->fncHandle /* method */,
-                                                           pParam->exactContextHnd /* context */,
-                                                           TRUE /* speculative */);
+                                                           pParam->exactContextHnd /* context */);
```

Comment:
The inline observation just below should be updated to `CLASS_INIT_FAILURE`.

And since that was the only use of `CLASS_INIT_FAILURE_SPEC` we can remove that entry from inline.def.